python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -83,7 +84,7 @@ class StandardAutoscaler:
 
     def __init__(
             self,
-            config_path: str,
+            config_reader: Union[str, Callable[[], dict]],
```

Comment:
I'm a little uneasy about the naming and typing of this thing across this file and monitor.py

One alternative is to 
leave config_path as a str and to 
add an additional optional config_reader arg.

If config_reader is supplied, use that to read configs and log a message that config_path is being ignored...

Ok, maybe that's not so great either.

Anyways, logging slight unease.

Reply:
shouldn't this be `Callable[List, Dict]`?

Reply:
`Callable[[], Dict]` or `Callable[[], dict]` because it takes in no args and produces a dict.

Reply:
Let me add a TODO to drop support for bare "str"--- it's just a big change to many files, so best to avoid for now.